docs/utime.rst

```diff
@@ -0,0 +1,159 @@
+    specifically, ring) arithmetics to produce correct result even for 
+    wrap-around values (as long as they not too distant inbetween, see below).
+    The function returns **signed** value in the range 
+    *[-TICKS_PERIOD/2 .. TICKS_PERIOD/2-1]* (that’s a typical range definition 
+    for two’s-complement signed binary integers). If the result is negative, it
+    means that *ticks1* occurred earlier in time than *ticks2*. Otherwise, it 
+    means that *ticks1* occurred after *ticks2*. This holds **only** if 
+    *ticks1* and *ticks2* are apart from each other for no more than 
+    *TICKS_PERIOD/2-1* ticks. If that does not hold, incorrect result will be
+    returned. Specifically, if two tick values are apart for *TICKS_PERIOD/2-1*
+    ticks, that value will be returned by the function. However, if
+    *TICKS_PERIOD/2* of real-time ticks has passed between them, the function 
+    will return *-TICKS_PERIOD/2* instead, i.e. result value will wrap around 
+    to the negative range of possible values.
+
+    Informal rationale of the constraints above: Suppose you are locked in a 
+    room with no means to monitor passing of time except a standard 12-notch 
+    clock. Then if you look at dial-plate now, and don’t look again for another
+    13 hours (e.g., if you fall for a long sleep), then once you finally look 
+    again, it may seem to you that only 1 hour has passed. To avoid this 
+    mistake, just look at the clock regularly. Your application should do the 
+    same. “Too long sleep” metaphor also maps directly to application behavior:
+    don’t let your application run any single task for too long. Run tasks in 
+    steps, and do time-keeping inbetween.
+
+    :func:`utime.ticks_diff()` is designed to accommodate various usage 
+    patterns, among them:
+
+
+    Polling with timeout. In this case, the order of events is known, and you
+    will deal only with positive results of :func:`utime.ticks_diff()`:
+
+    .. code-block:: python
+
+        # Wait for GPIO pin to be asserted, but at most 500us
+        start = time.ticks_us()
+        while pin.value() == 0:
+            if time.ticks_diff(time.ticks_us(), start) > 500:
+                raise TimeoutError
+
+
+    Scheduling events. In this case, :func:`utime.ticks_diff()` result may be
+    negative if an event is overdue:
+
+
+    .. code-block:: python
+
+        # This code snippet is not optimized
+        now = time.ticks_ms()
+        scheduled_time = task.scheduled_time()
+        if ticks_diff(scheduled_time, now) > 0:
+            print("Too early, let's nap")
+            sleep_ms(ticks_diff(scheduled_time, now))
+            task.run()
+        elif ticks_diff(scheduled_time, now) == 0:
+            print("Right at time!")
+            task.run()
+        elif ticks_diff(scheduled_time, now) < 0:
+            print("Oops, running late, tell task to run faster!")
+            task.run(run_faster=true)
```

Comment:
A lot of the description in this page are a bit hard to follow, perhaps it would be better to keep the short descriptions (the first sentence) and then link to the official MicroPython docs for that methods for more info.